### DISCUSSION TITLE
Complex result of Maxent AC

### DISCUSSION THREAD

**Discussion by Lanlin123** [TRIQS/solid_dmft#86] [Q&A]
Dear TRIQS developers,

When I do a Maxent AC calculation, I get a warning like this:
![image](https://github.com/user-attachments/assets/54c6761b-8d67-49f8-b3ac-a29a236f55ff)
And I get  unreasonable k resolved spectral function using this AC self energy. I don't know whether the problem is due to the Maxent parameter setting or DMFT calculation. Can you give me some advice? Thank you.

**Comment by the-hampel** [comment]
Hi @Lanlin123 ,
can you give a bit more details on how you called the AC (please paste the function call)? It seems that you are using a lot of alpha values ~800? Typically ~30 should be sufficient. At it probably a good hint that results might be problematic since they are complex. That should only be case if you have a system with spin-orbit coupling or similar. Is that the case? 

Best,
Alex

**Reply by Lanlin123** [reply]
Hi Alex,

Below is the AC function setting I used:
_Sigma_real_freq = sigma_maxent.main(external_path='./maxent.h5',
                                    omega_min=-10, omega_max=10,
                                    maxent_error=0.01, iteration=None,
                                    n_points_maxent=1000,
                                    n_points_alpha=800,
                                    analyzer='LineFitAnalyzer',
                                    n_points_interp=4000,
                                    n_points_final=4000,
                                    continuator_type='inversion_sigmainf')[0][0]_
I don't consider spin-orbit coupling in this setting. I have tried the Pade method and it can give a reasonable k resolved spectral function. But I just want to use Maxent to do the AC calculation. Could you please give me some advice?

**Reply by the-hampel** [reply]
Yes. I think this setting you choose will take quite long to run?! Anyway, start with the following setting: 
```
omega_min=-15, omega_max=15,
maxent_error=0.05, iteration=None,
n_points_maxent=101,
n_points_alpha=20,
analyzer='LineFitAnalyzer',
n_points_interp=3001,
n_points_final=3001,
continuator_type='inversion_sigmainf')[0][0]
```
This will set you number of frequencies on which the AC is performed to `n_points_maxent=101` on a hyperbolic mesh, i.e. much denser around 0 and coarse further aways. That should be sufficient for starters. Then alpha points are 20. The end points are always the same. For getting a "feeling" for the AC just use a few points and check whether the optimal alpha value reported by the script is somewhere in the second half, i.e. between 10 and 20 now. If not change the `maxent_error` parameter until the alpha value is in between these values. Can you try that and report back to me what the chosen alpha value of the script is and how the self-energy of one of the orbitals looks like? Thanks.

**Reply by Lanlin123** [reply]
Hi Alex,
I used the parameter setting you provided. it works and is much faster than before.  I have solved my problem. It is due to the unreasonble structure I used. Thank you so much.
Best regards,
Lanlin.
